float64 loc_x
float64 loc_y
float64 loc_z
int64 dir
int64 id
float64 pos_x
float64 pos_y
float64 pos_z


